### PR TITLE
{EventHub} `az eventhubs namespace application-group create/delete/list/show/update` `az eventhubs namespace application-group policy add/remove`: Migrating eventhub application-group cmdlets to AAZ

### PR DESCRIPTION
**Pull request by schaudhari6254888**
**Related command**
`az eventhubs namespace application-group policy remove`

**Description**
This PR is about Migrating `az eventhubs namespace application-group` cmdlets to AAZ.
Also adds the breaking change for `az eventhubs namespace application-group policy remove` by removing parameters metric-id, rate-limit-threshold from --throttling-policy-config object.
Additionally changing `throttling-policy-config` name to `policy`, for `az eventhubs namespace application-group policy remove` command.

**Testing Guide**

**History Notes**

[EventHub] BREAKING CHANGE: `az eventhubs namespace application-group policy remove`: Renamed `--throttling-policy-config` to `--policy` and removed `metric-id` and `rate-limit-threshold` properties in it

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
EventHub

File: src/azure-cli/azure/cli/command_modules/eventhubs/Operation/AppgroupCustomFile.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        for col in throttling_policy_config:
+            policy_object.append(create_app_group_policy_object(col))
+        command_args_dict["policies"] = policy_object
+
+    command_args_dict.update({
+        "resource_group": resource_group_name,
+        "namespace_name": namespace_name,
+        "application_group_name": application_group_name
+    })
+    print(command_args_dict)
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/eventhubs/Operation/AppgroupCustomFile.py:51]: Why print it?

File: src/azure-cli/azure/cli/command_modules/eventhubs/Operation/AppgroupCustomFile.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    })
+    policy_object = []
+    for i in policy:
+        semaphor = 0
+        for j in application_group["policies"]:
+            if i["name"] == j["name"]:
+                application_group["policies"].remove(j)
+                semaphor = 1
+        if semaphor == 0:
+            raise CLIError('The following policy was not found: Name: ' + i["name"])
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/eventhubs/Operation/AppgroupCustomFile.py:96]: CLIError is not used. Please use other error types in this file https://github.com/Azure/azure-cli/blob/57577782547064257d69da916683a46e7c25e1af/src/azure-cli-core/azure/cli/core/azclierror.py

**Review by kairu-ms — CHANGES_REQUESTED** [review]
Please change the filename in snakecase.

**Comment by damodaravadhani** [comment]
Hide "--policies" in update command. Customer should not be able to use update to set policies.

**Comment by yonzhan** [comment]
Please fix CI issues.

**Comment by schaudhari6254888** [comment]
> Please change the filename in snakecase.

Done

File: src/azure-cli/azure/cli/command_modules/eventhubs/_help.py [hunk 3]
```diff
@@ -768,7 +768,7 @@
 type: command
 short-summary: Removes an application group policy from the existing policies. This cmdlet can be used to remove one or more throttling policies.
 examples:
-  - name: Removes a throttling policy from an applicatin group myAppGroup.
+  - name: Removes a throttling policy from an application group myAppGroup.
     text: |
-        az eventhubs namespace application-group policy remove --namespace-name mynamespace -g MyResourceGroup --name myAppGroup --throttling-policy-config name=policy1 metric-id=OutgoingMessages rate-limit-threshold=10500
+        az eventhubs namespace application-group policy remove --namespace-name mynamespace -g MyResourceGroup --name myAppGroup --policy name=policy1
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/eventhubs/_help.py:773]: The linter issue seems caused by this change in example. Missing required properties for the --policy argument.

Comment (schaudhari6254888) [reply src/azure-cli/azure/cli/command_modules/eventhubs/_help.py:773]: > The linter issue seems caused by this change in example. Missing required properties for the --policy argument.

We are removing the policy by it's name only.

File: src/azure-cli/azure/cli/command_modules/eventhubs/action.py [hunk 4]
```diff
@@ -109,38 +109,71 @@ def __call__(self, parser, namespace, values, option_string=None):
...
                 else:
                     raise CLIError('Only allowed values for metric_id are: {0}, {1}, {2}, {3}'.format(INCOMING_MESSAGES, INCOMING_BYTES, OUTGOING_MESSAGES, OUTGOING_BYTES))
 
             else:
                 raise InvalidArgumentValueError("Invalid Argument for:'{}' Only allowed arguments are 'name, rate-limit-threshold and metric-id'".format(option_string))
 
-        if (name is None) or (metric_id is None) or (rate_limit_threshold is None):
-            raise CLIError('One of the throttling policies is missing one of these parameters: name, metric-id, rate-limit-threshold')
+        if (throttling_policy["name"] is None) or (throttling_policy["throttling_policy"].get("rate_limit_threshold", None) is None) or (throttling_policy["throttling_policy"].get("metric_id", None) is None):
+            raise RequiredArgumentMissingError('Throttling policies is missing one of these parameters:name, metric-id, rate-limit-threshold')
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/eventhubs/action.py:156]: The linter error is raised from this line.

Comment (schaudhari6254888) [reply src/azure-cli/azure/cli/command_modules/eventhubs/action.py:153]: Solved.

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**